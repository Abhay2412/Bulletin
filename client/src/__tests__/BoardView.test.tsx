import React from "react";
import { fireEvent, render as reactRender } from "@testing-library/react";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import { createMemoryHistory } from "history";
import BoardView from "../pages/BoardViews/BoardsView";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("testing <Header /> component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <BoardView />
      </BrowserRouter>
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <BoardView />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should have add board icon", () => {
    expect(wrapper.find("#add-board-icon").exists()).toBeTruthy();
  });

  it("redirect to create board page", () => {
    const history = createMemoryHistory({
      initialEntries: ["/boardsView"],
    });
    const { getByTestId } = reactRender(
      <Provider store={store}>
        <BrowserRouter>
          <BoardView />
        </BrowserRouter>
      </Provider>
    );
    expect(history.location.pathname).toBe("/boardsView");
    fireEvent.click(getByTestId("add-board-button"));
  });
});
