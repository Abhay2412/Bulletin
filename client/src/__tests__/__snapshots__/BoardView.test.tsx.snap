// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Header /> component should render properly 1`] = `
<div
  className="MuiContainer-root css-1e57c4r-MuiContainer-root"
>
  <div
    className="boardHeaderBox MuiBox-root css-0"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-x4wfr5-MuiTypography-root"
    >
      Your Boards
    </h6>
    <div>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Filter By
      </button>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Sort
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 boardViewsBox css-1ld3b9g-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3 css-v3c8vo-MuiGrid-root"
    >
      <div
        className="board-icon-container"
        id="add-board-icon"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-16xo46n-MuiButtonBase-root-MuiButton-root"
          data-testid="add-board-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="Add A New Board"
            src="newBoard.png"
            style={
              Object {
                "boxShadow": "1px 5px 8px rgba(165, 165, 165)",
              }
            }
            width="80%"
          />
        </button>
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1x7nvub-MuiTypography-root"
        >
          Create A New Board
        </h6>
      </div>
    </div>
  </div>
</div>
`;
